systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,299 @@
+     * | | Topic 1     |       | Topic 2     |     ...     | Topic N     |           |
+     * | | +---------+ |       | +---------+ |             | +---------+ |           |
+     * | | | Thread  | |       | | Thread  | |             | | Thread  | |           |
+     * | | |         | |       | |         | |             | |         | |           |
+     * | | | Creation| |       | | Creation| |             | | Creation| |           |
+     * | | | Update  | |       | | Update  | |             | | Update  | |           |
+     * | | | Deletion| |       | | Deletion| |             | | Deletion| |           |
+     * | | +---------+ |       | +---------+ |             | +---------+ |           |
+     * | +-------------+       +-------------+             +-------------+           |
+     * +-----------------------------------------------------------------------------+
+     *
+     * @param testStorage           An instance of TestStorage containing configuration and state needed for topic operations.
+     * @param numberOfTopics        The number of Kafka topics to be processed.
+     * @param spareEvents           The number of spare events to be consumed during the process.
+     * @param warmUpTasksToProcess  The number of tasks to warm-up performance and optimize JIT. This number is used just for offsetting.
+     *
+     * @return                      The total time taken to complete all topic lifecycles in milliseconds.
+     */
+    public static long processAllTopicsConcurrently(TestStorage testStorage, int numberOfTopics, int spareEvents, int warmUpTasksToProcess) {
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        ExtensionContext extensionContext = ResourceManager.getTestContext();
+
+        long startTime = System.nanoTime();
+
+        for (int topicIndex = warmUpTasksToProcess; topicIndex < numberOfTopics + warmUpTasksToProcess; topicIndex++) {
+            final int finalTopicIndex = topicIndex;
+            CompletableFuture<Void> future = CompletableFuture.runAsync(() -> performFullLifecycle(finalTopicIndex, testStorage, extensionContext), EXECUTOR);
+            futures.add(future);
+        }
+
+        // consume spare events
+        for (int j = 0; j < spareEvents; j++) {
+            futures.add(j, CompletableFuture.completedFuture(null));
+        }
+
+        // Wait for all topics to complete their lifecycle
+        CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).join();
+        LOGGER.info("All topic lifecycles completed.");
+
+        long allTasksTimeMs = (System.nanoTime() - startTime) / 1_000_000;
+
+        if (warmUpTasksToProcess != 0) {
+            // boundary between tests => less likelihood that tests would influence each other
+            LOGGER.info("Cooling down");
+            try {
+                Thread.sleep(30_000);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
```

Comment:
It's fine that you are giving some space between the tests, but I'm not sure I understand how they can influence each other. You have deployed one Kafka cluster and you are just deploying the KafkaTopics with different configurations, so UTO would have some kind of issue with that? The magical number there is not helping much as well.
And `Cooling down` is not much saying TBH

Reply:
So as @fvaleri suggested with a smaller checking interval would it make sense? I am also not sure how much it helps cause mine is a little bit different from yours (i.e., you have a shared Kafka cluster where you do creation/modification and deletion) but in this case, we deploy a brand new Kafka cluster. So if that still makes sense I can make it to 10 seconds or 5 seconds.

Reply:
If it's needed, fine let's have some lower duration of this sleep. But I would definitely change the log message.

Reply:
Updated to 5 seconds